pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantCannotSeeNonTenantStats failed

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantCannotSeeNonTenantStats [failed](https://mesolite.cluster.engflow.com/invocations/default/34bc213b-e689-4201-b468-9b8acf33bad3?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbC9zdGF0dXNjY2w6c3RhdHVzY2NsX3Rlc3Q=) on release-25.3 @ [048689ed51574b12f3da77fa9dc679a592c98672](https://github.com/cockroachdb/cockroach/commits/048689ed51574b12f3da77fa9dc679a592c98672):


```
=== RUN   TestTenantCannotSeeNonTenantStats
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantCannotSeeNonTenantStats3001365606
    test_log_scope.go:76: use -show-logs to present logs inline
    tenant_status_test.go:565: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantCannotSeeNonTenantStats3001365606
--- FAIL: TestTenantCannotSeeNonTenantStats (71.42s)
=== RUN   TestTenantCannotSeeNonTenantStats/tenant-stats
    tenant_status_test.go:542: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/tenant_status_test.go:537
        	            				pkg/ccl/serverccl/statusccl/tenant_status_test.go:542
        	Error:      	Not equal: 
        	            	expected: []string{"CREATE DATABASE roachblog_t", "CREATE TABLE posts_t (id INT8 PRIMARY KEY, body STRING)", "INSERT INTO posts_t VALUES (_, __more__)", "SELECT * FROM posts_t", "SET database = roachblog_t"}
        	            	actual  : []string{"CREATE DATABASE roachblog_t", "CREATE TABLE posts_t (id INT8 PRIMARY KEY, body STRING)", "SELECT * FROM posts_t", "SET database = roachblog_t"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,5 +1,4 @@
        	            	-([]string) (len=5) {
        	            	+([]string) (len=4) {
        	            	  (string) (len=27) "CREATE DATABASE roachblog_t",
        	            	  (string) (len=55) "CREATE TABLE posts_t (id INT8 PRIMARY KEY, body STRING)",
        	            	- (string) (len=40) "INSERT INTO posts_t VALUES (_, __more__)",
        	            	  (string) (len=21) "SELECT * FROM posts_t",
        	Test:       	TestTenantCannotSeeNonTenantStats/tenant-stats
--- FAIL: TestTenantCannotSeeNonTenantStats/tenant-stats (0.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #151867 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantCannotSeeNonTenantStats failed [C-test-failure O-robot T-observability branch-release-25.2.5-rc release-blocker]
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantCannotSeeNonTenantStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53694

--- a/pkg/ccl/serverccl/statusccl/tenant_status_test.go
+++ b/pkg/ccl/serverccl/statusccl/tenant_status_test.go
@@ -402,12 +402,16 @@ func TestTenantCannotSeeNonTenantStats(t *testing.T) {
 	skip.UnderStressWithIssue(t, 113984)
 
 	ctx := context.Background()
+	sqlStatsKnobs := sqlstats.CreateTestingKnobs()
+	sqlStatsKnobs.SynchronousSQLStats = true
 	testCluster := serverutils.StartCluster(t, 3 /* numNodes */, base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
 			Knobs: base.TestingKnobs{
+				SQLStatsKnobs: sqlStatsKnobs,
 				SpanConfig: &spanconfig.TestingKnobs{
 					ManagerDisableJobCreation: true, // TODO(irfansharif): #74919.
-				}},
+				},
+			},
 			DefaultTestTenant: base.TestControlsTenantsExplicitly,
 		},
 	})
@@ -418,7 +422,7 @@ func TestTenantCannotSeeNonTenantStats(t *testing.T) {
 	tenant, sqlDB := serverutils.StartTenant(t, server, base.TestTenantArgs{
 		TenantID: roachpb.MustMakeTenantID(10 /* id */),
 		TestingKnobs: base.TestingKnobs{
-			SQLStatsKnobs: sqlstats.CreateTestingKnobs(),
+			SQLStatsKnobs: sqlStatsKnobs,
 		},
 	})
 
@@ -478,7 +482,7 @@ func TestTenantCannotSeeNonTenantStats(t *testing.T) {
 	}
 
 	conn = sqlutils.MakeSQLRunner(systemLayer.SQLConn(t))
-	sqlstatstestutil.WaitForStatementEntriesAtLeast(t, conn, len(testCaseTenant), sqlstatstestutil.StatementFilter{
+	sqlstatstestutil.WaitForStatementEntriesAtLeast(t, conn, len(testCaseNonTenant), sqlstatstestutil.StatementFilter{
 		App: appName,
 	})
 
@@ -524,11 +528,6 @@ func TestTenantCannotSeeNonTenantStats(t *testing.T) {
 
 		var actualStatements []string
 		for _, respStatement := range actual.Statements {
-			if respStatement.Stats.FailureCount > 0 {
-				// We ignore failed statements here as the INSERT statement can fail and
-				// be automatically retried, confusing the test success check.
-				continue
-			}
 			if respStatement.Key.KeyData.App != appName {
 				continue
 			}

--- a/pkg/server/application_api/sql_stats_test.go
+++ b/pkg/server/application_api/sql_stats_test.go
@@ -464,11 +464,6 @@ func TestStatusAPIStatements(t *testing.T) {
 		// See if the statements returned are what we executed.
 		var statementsInResponse []string
 		for _, respStatement := range resp.Statements {
-			if respStatement.Stats.FailureCount > 0 {
-				// We ignore failed statements here as the INSERT statement can fail and
-				// be automatically retried, confusing the test success check.
-				continue
-			}
 			if respStatement.Key.KeyData.App != "test" {
 				continue
 			}
@@ -807,14 +802,7 @@ func TestStatusAPICombinedStatementsWithFullScans(t *testing.T) {
 		// statement statistics received from the server response.
 		actualResponseStatsMap := make(map[string]serverpb.StatementsResponse_CollectedStatementStatistics)
 		for _, respStatement := range resp.Statements {
-			// Skip failed statements: The test app may encounter transient 40001
-			// errors that are automatically retried. Thus, we only consider
-			// statements that were that were successfully executed by the test app
-			// to avoid counting such failures. If a statement that we expect to be
-			// successful is not found in the response, the test will fail later.
-			if respStatement.Key.KeyData.App == testAppName && respStatement.Stats.FailureCount == 0 {
-				actualResponseStatsMap[respStatement.Key.KeyData.Query] = respStatement
-			}
+			actualResponseStatsMap[respStatement.Key.KeyData.Query] = respStatement
 		}
 
 		for respQuery, expectedData := range expectedStatementStatsMap {
@@ -927,11 +915,6 @@ func TestStatusAPICombinedStatements(t *testing.T) {
 		var statementsInResponse []string
 		expectedTxnFingerprints := map[appstatspb.TransactionFingerprintID]struct{}{}
 		for _, respStatement := range resp.Statements {
-			if respStatement.Stats.FailureCount > 0 {
-				// We ignore failed statements here as the INSERT statement can fail and
-				// be automatically retried, confusing the test success check.
-				continue
-			}
 			if respStatement.Key.KeyData.App != "test" {
 				// CombinedStatementStats should filter out internal queries.
 				if strings.HasPrefix(respStatement.Key.KeyData.Query, catconstants.InternalAppNamePrefix) {

